core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala

```diff
@@ -122,7 +122,8 @@ private[netty] class NettyRpcEnv(
 
   @Nullable
   override lazy val address: RpcAddress = {
-    if (server != null) RpcAddress(host, server.getPort()) else null
+    if (server != null)
+      RpcAddress(host, conf.getInt("spark.driver.advertisedPort", server.getPort())) else null
```

Comment:
My assumption was that this was the safest behaviour for the Netty based rpc. If the `spark.driver.advertisedPort` setting is `0`, it's an operator error, no different than setting the value to, say, `1`. Not sure if there is any value in testing for `0` here. Ideas welcome.